RS/JX/Rule12-1

Jiarongx-Xie opened the pull request:
This rule has not been tested.

weilixu commented on rct229/rulesets/ashrae9012019/section12/section12rule1.py:
```diff
@@ -11,35 +14,88 @@ class Section12Rule1(RuleDefinitionListIndexedBase):
+                misc_equipment_power_b = misc_equipment_b["power"]
+                misc_equipment_power_p = misc_equipment_p["power"]
+                if misc_equipment_power_b > misc_equipment_power_p:
+                    reduced_misc_equipment_power.append(misc_equipment_power_b)
```
The RDS says appending an object following the format below:
```python
{"id": misc_equipment_b.id, "baseline_power": misc_equipment_power_b, "proposed_power": misc_equipment_power_p}
```

weilixu commented on rct229/rulesets/ashrae9012019/section12/section12rule1.py:
```diff
@@ -11,35 +14,88 @@ class Section12Rule1(RuleDefinitionListIndexedBase):
+            ]
+            compliance_path = calc_vals["compliance_path"]
+            return len(unexpected_misc_equipment_power) == 0 and (
+                compliance_path is None or compliance_path != "CODE_COMPLIANT"
```
` compliance_path is None` is part of `compliance_path != "CODE_COMPLIANT"` logic.
So `compliance_path = None, compliance_path != "CODE_COMPLIANCE" is True`.

weilixu replied:
The `CODE_COMPLIANCE` shall be a schema enum.

At the top of this script,
```python
COMPLIANCE_PATH_TYPE = SchemaEnums.schema_enums["CompliancePathOptions2019ASHRAE901"]
```
Then in here,  you can do:
`COMPLIANCE_PATH_TYPE.CODE_COMPLIANCE`

weilixu commented:
@Jiarongx-Xie Thanks for addressing the comments.
There is one more comment left and a conflicting files need to be addressed.

Jiarongx-Xie commented:
@weilixu Thanks for the review. I have addressed all comments and all tests passed when tested locally. It seems that the CI failed due to some problem not related to this rule.

weilixu commented on rct229/rulesets/ashrae9012019/section12/section12rule1.py:
```diff
@@ -11,35 +17,95 @@ class Section12Rule1(RuleDefinitionListIndexedBase):
+                misc_equipment_p,
+            ) in proposed_baseline_misc_equipment_pairs:
+                misc_equipment_power_b = misc_equipment_b.get("power", ZERO.POWER)
+                misc_equipment_power_p = misc_equipment_p.get("power", ZERO.POWER)
```
@JacksonJ-KC After reading this code, I wonder if we ever considered what means equal between `misc_equipment_b` and `misc_equipment_p`. Have we ever discussed whether the rule's scope is limited in `power` or the object? What if the `misc_equipment_b` has a different energy type but same power as `misc_equipment_p`, do they consider equal in this rule?

JacksonJ-KC replied:
Good point Weili, I don't think this came up during discussion in the RDS meetings. I think the energy type should also need to match for the rule to pass.

weilixu reviewed (approved):
Good work.
I have one question for RDS logic - but others look great. Thank you.
